src/otg/dataset/study_index.py

```diff
@@ -477,3 +477,113 @@ def from_source(
                 )
             )
         )
+
+
+@dataclass
+class StudyIndexUKBiobank(StudyIndex):
+    """Study index dataset from UKBiobank.
+
+    The following information is extracted:
+
+    - studyId
+    - pubmedId
+    - publicationDate
+    - publicationJournal
+    - publicationTitle
+    - publicationFirstAuthor
+    - traitFromSource
+    - ancestry_discoverySamples
+    - ancestry_replicationSamples
+    - initialSampleSize
+    - nCases
+    - replicationSamples
+
+    Some fields are populated as constants, such as projectID, studyType, and initial sample size.
+    """
+
+    @classmethod
+    def from_source(
+        cls: type[StudyIndexUKBiobank],
+        ukbiobank_studies: DataFrame,
+    ) -> StudyIndexUKBiobank:
+        """This function ingests study level metadata from UKBiobank.
+
+        Args:
+            ukbiobank_studies (DataFrame): UKBiobank study manifest file loaded in spark session.
+
+        Returns:
+            StudyIndexUKBiobank: Annotated UKBiobank study table.
+        """
+        return cls(
+            _df=(
+                ukbiobank_studies.select(
+                    f.col("code").alias("studyId"),
+                    f.col("trait").alias("traitFromSource"),
+                    f.col("n_cases").cast("long").alias("nCases"),
+                    f.col("n_total").cast("string").alias("initialSampleSize"),
+                    f.col("in_path").alias("summarystatsLocation"),
+                )
+                # Replace any trait double spaces with single.
+                .withColumn(
+                    "traitFromSource",
+                    f.regexp_replace(f.col("traitFromSource"), r" +", " "),
+                )
```

Comment:
As @ireneisdoomed and @d0choa once explained to me, it's usually a good idea to replace multiple `.withColumn` statements with a bigger `select` statment. So in this case, for example, I would change it like this:

```suggestion
                ukbiobank_studies.select(
                    f.col("code").alias("studyId"),
                    # Replace any trait double spaces with single.
                    f.regexp_replace(f.col("trait"), r" +", " ").alias("traitFromSource"),
                    f.col("n_cases").cast("long").alias("nCases"),
                    f.col("n_total").cast("string").alias("initialSampleSize"),
                    f.col("in_path").alias("summarystatsLocation"),
                )
```

The reason is that each `withColumn` statments adds a bit of computational complexity, whereas a big `select` is more efficient

Reply:
This would also apply to some (but not all) of the `withColumn` statements below. Essentially, if there's an easy way to move it to the `select` statement, do it; but if it seems tricky or even not possible (which it could be, in some cases), just leave them be